smartpower3/channel.cpp

```diff
@@ -15,18 +15,18 @@ Channel::Channel(TFT_eSPI *tft, TwoWire *theWire, uint16_t x, uint16_t y,
 	_volt = new FndWidget(tft);
 	_current = new FndWidget(tft);
 
-	icon_op = new IconWidget(tft, channel);
-	icon_sp = new IconWidget(tft, channel);
-	icon_tp = new IconWidget(tft, channel);
-	icon_tw = new IconWidget(tft, channel);
-	icon_ip = new IconWidget(tft, channel);
+	icon_op = new ProtectionChannelIconWidget(tft, channel, PROTECTION_OVERVOLTAGE);  //0
```

Comment:
IconWidget was changed to mostly individual icon classes for the following reasons:
- allows inheritance and overloading (behavior can be customized per icon type)
- some settings moved to class constructor + ints changed to enums
- prevents tons of various drawing methods

Basically, I've ran into a roadblock when I tried to mark multiple icons on screen.